examples/cluster/plot_hdbscan.py

```diff
@@ -0,0 +1,225 @@
+# -*- coding: utf-8 -*-
+"""
+====================================
+Demo of HDBSCAN clustering algorithm
+====================================
+
+In this demo we will take a look at :class:`sklearn.cluster.HDBSCAN` from the
+perspective of generalizing the :class:`sklearn.cluster.DBSCAN` algorithm.
+We'll compare both algorithms on specific datasets. Finally we'll evaluate
+HDBSCAN's sensitivity to certain hyperparameters. We first define a couple
+utility functions for convenience.
+"""
+
+import numpy as np
+
+from sklearn.cluster import HDBSCAN, DBSCAN
+from sklearn import metrics
+from sklearn.datasets import make_blobs
+import matplotlib.pyplot as plt
+
+
+def plot(X, labels=None, probabilities=None, kwargs=None, ground_truth=False):
+    labels = labels if labels is not None else np.ones(X.shape[0])
+    probabilities = probabilities if probabilities is not None else np.ones(X.shape[0])
+    _, ax = plt.subplots()
+    # Black removed and is used for noise instead.
+    unique_labels = set(labels)
+    colors = [plt.cm.Spectral(each) for each in np.linspace(0, 1, len(unique_labels))]
+    # The probability of a point belonging to its labeled cluster determines
+    # the size of its marker
+    proba_map = {idx: probabilities[idx] for idx in range(len(labels))}
+    for k, col in zip(unique_labels, colors):
+        if k == -1:
+            # Black used for noise.
+            col = [0, 0, 0, 1]
+
+        class_index = np.where(labels == k)[0]
+        for ci in class_index:
+            ax.plot(
+                X[ci, 0],
+                X[ci, 1],
+                "x" if k == -1 else "o",
+                markerfacecolor=tuple(col),
+                markeredgecolor="k",
+                markersize=4 if k == -1 else 1 + 5 * proba_map[ci],
+            )
+    n_clusters_ = len(set(labels)) - (1 if -1 in labels else 0)
+    preamble = "True" if ground_truth else "Estimated"
+    title = f"{preamble} number of clusters: {n_clusters_}"
+    if kwargs is not None:
+        title += f" | {kwargs=}"
+    ax.set_title(title)
+    plt.show()
+
+
+def print_scores(labels, labels_true, kwargs):
```

Comment:
I feel that printing out these metrics can distract the reader from the other messages in this example.

Reply:
Yeah I had based this off of the `DBSCAN` plotting demo, but the nature of the demonstration for `HDBSCAN` has changed since then and I think is more focused on qualitative demonstration rather than comparing these quantitative metrics. Will remove.